master/internal/rm/kubernetesrm/resource_pool.go

```diff
@@ -107,22 +98,17 @@ func (k *kubernetesResourcePool) ResourcesReleased(msg sproto.ResourcesReleased)
 	k.resourcesReleased(msg)
 }
 
-func (k *kubernetesResourcePool) UpdatePodStatus(msg sproto.UpdatePodStatus) {
+func (k *kubernetesResourcePool) JobSchedulingStateChanged(msg jobSchedulingStateChanged) {
 	k.mu.Lock()
 	defer k.mu.Unlock()
 	k.reschedule = true
 
-	id, ok := k.containerIDtoAllocationID[msg.ContainerID]
-	if !ok {
-		return
-	}
-
 	for it := k.reqList.Iterator(); it.Next(); {
 		req := it.Value()
-		if req.AllocationID == id {
+		if req.AllocationID == msg.AllocationID {
 			req.State = msg.State
 			if sproto.ScheduledStates[req.State] {
-				k.allocationIDToRunningPods[id]++
+				k.allocationIDToRunningPods[msg.AllocationID] += msg.NumPods
```

Comment:
does the job know how many pods it has running, it is a little tragic that we need to keep track of this map

feel free to just make a follow up ticket or ignore any of these

Reply:
I agree, tragic but no it doesn't know how many are "running" where our definition of "running" is post scheduling and bound to a node.